### ISSUE TITLE
v3 .row add flex-wrap style breaks in safari

### ISSUE THREAD

**Issue by Wendong-Wu** [twbs/bootstrap#18306]
In safari, if bootstrap rows are added the following class:
**css**:
```css
.row-flex {
  display: -webkit-flex;
  -webkit-flex-wrap: wrap;
}
```

**html**:
```html
<div class="row row-flex">
    <div class="col-sm-4"></div>
    <div class="col-sm-8"></div>
</div>
```
[codepen demo](http://codepen.io/Wendong-Wu/pen/PPvzbz)
The children columns that are add up to 12 will break into two lines. This is caused by pseudo elements "before" and "after" in `.row`. The `content: ' '`  will have an impact on width in flexbox in Safari. But if we change the css to this, the problem will be fixed. 
**css**:
```css
.row-flex:before, .row-flex:after {
  width: 0;
}
```

However, the pseudo elements still occupy some tiny space in Safari's flexbox even if it is set to width 0. But I guess this is a Safari bug. 

Though this is a rare use case, I think adding `width: 0` to the pseudo elements in `.row` will be safer choice.

**Comment by cvrebert** [comment]
@Wendong-Wu Is the "v3" in the title a typo?

**Comment by Wendong-Wu** [comment]
@cvrebert no I just indicate this issue is for Bootstrap 3. Because V4 use flexbox to implement the row, so it may have avoided this problem already.

**Comment by Wendong-Wu** [comment]
Added [codepen demo](http://codepen.io/Wendong-Wu/pen/PPvzbz) to show the issue.

**Comment by mdo** [comment]
v3 doesn't support flexbox, and that class isn't present in our code. If you're adding those styles yourself, it'd make sense you'd have to add more CSS to fix any potential bugs that stem from it.

**Comment by quicksketch** [comment]
We (re)discovered this issue in Bootstrap 4. The symptoms are exactly as described in the original post. Two columns within a `.row` or `.container` class will exhibit this bug in Safari 9.x (but not in Safari 8.x, which is fine) so users of El Capitan will see this.

> However, the pseudo elements still occupy some tiny space in Safari's flexbox even if it is set to width 0. But I guess this is a Safari bug. 

We have also confirmed this particular problem. Setting `width: 0` on the pseudo-element doesn't fix the problem. We were able to solve this problem in Safari only by removing the `:before` pseudo-element entirely. According to the inline documentation, that means that it will affect Opera when contenteditable elements exist, but in our situation that is minor compared to our layouts being broken for all El Capitan Safari users.

So perhaps we could reopen this issue and label it for v4?
